spec/operators/groupBy-spec.js

```diff
@@ -61,4 +76,449 @@ describe('Observable.prototype.groupBy()', function () {
+      d: 'foO ',
+      e: ' Baz   ',
+      f: '  qux ',
+      g: '   bar',
+      h: ' BAR  ',
+      i: 'FOO ',
+      j: 'baz  ',
+      k: ' bAZ ',
+      l: '    fOo    '
+    };
+    var e1 = hot('-1--2--^-a-b-c-d-e-f-g-h-i-j-k-l-|', values);
+    var expected =      '--w---x---y-z-------------|';
+    var w =             '--a-b---d-';
+    var unsubw =        '         !';
+    var x =             '------c-------g-h---------|';
+    var y =             '----------e---------j-k---|';
+    var z =             '------------f-------------|';
+
+    var expectedGroups = {
+      w: Rx.TestScheduler.parseMarbles(w, values),
+      x: Rx.TestScheduler.parseMarbles(x, values),
+      y: Rx.TestScheduler.parseMarbles(y, values),
+      z: Rx.TestScheduler.parseMarbles(z, values)
+    };
+
+    var fooUnsubscriptionFrame = Rx.TestScheduler.getUnsubscriptionFrame(unsubw);
+
+    var source = e1
+      .groupBy(function (val) {
+        return val.toLowerCase().trim();
+      })
+      .map(function (group) {
+        var arr = [];
+
+        var subscription = group
+          .materialize()
+          .map(function (notification) {
+            return { frame: rxTestScheduler.frame, notification: notification };
+          })
+          .subscribe(function (value) {
+            arr.push(value);
+          });
+
+        if (group.key === 'foo') {
+          rxTestScheduler.schedule(function () {
+            subscription.unsubscribe();
+          }, fooUnsubscriptionFrame - rxTestScheduler.frame);
+        }
+        return arr;
+      });
+
+    expectObservable(source).toBe(expected, expectedGroups);
+  });
+
+  it('should allow inners to be unsubscribed early at different times', function () {
+    var values = {
+      a: '  foo',
+      b: ' FoO ',
+      c: 'baR  ',
+      d: 'foO ',
+      e: ' Baz   ',
+      f: '  qux ',
+      g: '   bar',
+      h: ' BAR  ',
+      i: 'FOO ',
+      j: 'baz  ',
+      k: ' bAZ ',
+      l: '    fOo    '
+    };
+    var e1 = hot('-1--2--^-a-b-c-d-e-f-g-h-i-j-k-l-|', values);
+    var expected =      '--w---x---y-z-------------|';
+    var w =             '--a-b---d-';
+    var unsubw =        '         !';
+    var x =             '------c------';
+    var unsubx =        '            !';
+    var y =             '----------e------';
+    var unsuby =        '                !';
+    var z =             '------------f-------';
+    var unsubz =        '                   !';
+
+    var expectedGroups = {
+      w: Rx.TestScheduler.parseMarbles(w, values),
+      x: Rx.TestScheduler.parseMarbles(x, values),
+      y: Rx.TestScheduler.parseMarbles(y, values),
+      z: Rx.TestScheduler.parseMarbles(z, values)
+    };
+
+    var unsubscriptionFrames = {
+      foo: Rx.TestScheduler.getUnsubscriptionFrame(unsubw),
+      bar: Rx.TestScheduler.getUnsubscriptionFrame(unsubx),
+      baz: Rx.TestScheduler.getUnsubscriptionFrame(unsuby),
+      qux: Rx.TestScheduler.getUnsubscriptionFrame(unsubz)
+    };
```

Comment:
@blesh I settled for your type of solution in this case because the TestScheduler doesn't give us a way to prescribe when inner Observables are unsubscribed, only the outer one with `unsub` in `expectObservable(source, unsub)`. I thought this test is probably one-of-a-kind and it's not worth spending time on the TestScheduler for this.